pkgs/os-specific/linux/kernel/common-config.nix

```
@@ -373,6 +373,18 @@ let
       DRM_VC4_HDMI_CEC = yes;
     };
 
+    # Enables Rust support in the Linux kernel. This is currently not enabled by default, because it requires
+    # patching the Linux kernel for the specific Rust toolchain in nixpkgs. These patches usually take a bit
+    # of time to appear and this would hold up Linux kernel and Rust toolchain updates.
+    #
+    # Once Rust in the kernel has more users, we can reconsider enabling it by default.
+    rust = optionalAttrs ((features.rust or false)
+                          && versionAtLeast version "6.6"
+                          && stdenv.hostPlatform.system == "x86_64-linux") {
```

```suggestion
                          ) {
```

Since it's opt-in you can leave it up to the person opting-in to check this.

In general we shouldn't scatter these sorts of platform restrictions all over the place, and we shouldn't ever silently ignore a directive like this:
- use `meta.hydraPlatforms` if Hydra can't handle a situation, and
- `meta.badPlatforms` if you're 100% sure that it is totally impossible to build for the specified platform under all circumstances

---

Fixed.